fix: system tray compilation on 32-bit Windows

### Bug Description

 system_tray/windows.rs fails to compile when targeting 32-bit Windows (i686-pc-windows-msvc).

On Win32, `SetWindowLongPtrW` is a macro that expands to `SetWindowLongW`, which takes `i32` as the third parameter — not `isize` as on 64-bit. The current code casts to `isize`
 explicitly, causing a type mismatch on Win32.

Two call sites are affected:
- Line 168: setting the user data pointer (as isize)
- Line 266: clearing the user data pointer (literal 0 without cast)

Expected: compiles on both Win32 and Win64.
Actual: compilation error on Win32:

```shell
error[E0308]: mismatched types
     --> internal/core/items/system_tray/windows.rs:168
      |
      |     SetWindowLongPtrW(inner.hwnd, GWLP_USERDATA, &*inner as *const Inner as isize)
      |                                                                              ^^^^^
      |                                                   expected `i32`, found `isize`
```
Fix: replace as isize / literal 0 with as _, letting the compiler infer the correct integer type for both targets.

### Reproducible Code (if applicable)

```slint
N/A — this is a compilation error in Slint internals, not user code.
```

### Environment Details

- Slint Version: master (commit c13d8ca91)
- Platform/OS: Windows, cross-compiling for i686-pc-windows-msvc
- Programming Language: Rust
- Backend/Renderer: Skia

### Product Impact

 This issue blocks compilation on the 32-bit Windows
 target. The system tray feature is enabled via `SLINT_ENABLE_EXPERIMENTAL_FEATURES=1`.

--- a/internal/core/items/system_tray/windows.rs
+++ b/internal/core/items/system_tray/windows.rs
@@ -165,7 +165,7 @@ impl PlatformTray {
             hmenu: Cell::new(None),
             tip: RefCell::new(tip),
         });
-        unsafe { SetWindowLongPtrW(inner.hwnd, GWLP_USERDATA, &*inner as *const Inner as isize) };
+        unsafe { SetWindowLongPtrW(inner.hwnd, GWLP_USERDATA, &*inner as *const Inner as _) };
 
         Ok(Self { inner })
     }
@@ -263,7 +263,7 @@ impl Drop for PlatformTray {
 
             // Detach from the window before destroying it so any pending messages
             // resolve through DefWindowProc.
-            SetWindowLongPtrW(self.inner.hwnd, GWLP_USERDATA, 0);
+            SetWindowLongPtrW(self.inner.hwnd, GWLP_USERDATA, 0 as _);
             if let Some(m) = self.inner.hmenu.take() {
                 let _ = DestroyMenu(m);
             }

